### PR TITLE
Use cleaned up and modified code JavaScript of basic app features

### PR DESCRIPTION
**Pull request by raulcruise**
This pull request's goal is to update these files to include what is currently deployed as our MVP.

### CODE REVIEW

**Review by hollyyuqizheng — APPROVED** [review]

**Review by talvarez0507 — APPROVED** [review]

File: portfolio/src/main/webapp/calendar.js [hunk 1]
```diff
@@ -278,59 +290,105 @@ function getClosestNextHour(nextHour, workHourStartString, workHourEndString) {
...
-  return closestHour; 
+  return closestHour;
 }
- 
+
+/**
+ * Sets the default working hour start time based on user picked date
+ * and the current time.
+ * If user picked today to schedule for,
+ * The default working hour start time is the next closest hour,
```
Comment (rgcase) [inline portfolio/src/main/webapp/calendar.js]: It seems a little weird to me to display different working hours on different days. Can we not use the same working hours for "today" but then still just make sure we never select a time that's in the past? It just feels like it's going to be a little confusing for the user if they see their working hours change when they choose different days.

File: portfolio/src/main/webapp/calendar.js [hunk 2]
```diff
@@ -80,13 +80,21 @@ function createNewCalendarEventUserInput() {
     document.getElementById('new-event-name').value = EVENT_DEFAULT_NAME;
   }
 }
- 
+
+/** Shows the "Event added" header. */
+function showEventAddedHeader() {
+  const $eventListHeader = $('#event-added-header');
```
Comment (jamesnam-google) [inline portfolio/src/main/webapp/calendar.js]: You can merge these lines into,

$('#event-added-header').removeClass('d-none');

File: portfolio/src/main/webapp/calendar.js [hunk 3]
```diff
@@ -223,50 +233,52 @@ function setNewEventStartAndEndTimes() {
...
   $('#new-event-end-time').val(workHourEndString);
 }
 
 /** Checks if a date is today. */
 function isToday(date) {
   const today = new Date();
-  return (date.getFullYear() === today.getFullYear()
-      && date.getMonth() === today.getMonth()
-      && date.getDate() === today.getDate());
+  return (
```
Comment (jamesnam-google) [inline portfolio/src/main/webapp/calendar.js]: The outermost parens are not necessary.

File: portfolio/src/main/webapp/calendar.js [hunk 4]
```diff
@@ -278,59 +290,105 @@ function getClosestNextHour(nextHour, workHourStartString, workHourEndString) {
...
-/** 
- * Checks if the working hour start time is before end time. 
- * All arguments are of type integer. 
+/**
+ * Checks if the working hour start time is before end time.
+ * All arguments are of type integer.
  */
 function isWorkingHourValid(
-    workHourStartHour, workHourEndHour, workHourStartMinute, workHourEndMinute) {
+    workHourStartHour, workHourEndHour, workHourStartMinute,
```
Comment (jamesnam-google) [inline portfolio/src/main/webapp/calendar.js]: Consider declaring one param per line

File: portfolio/src/main/webapp/task.js [hunk 5]
```diff
@@ -230,13 +238,11 @@ function collectAllTasks() {
         parseInt(taskBody.childNodes[2].childNodes[2].childNodes[1].value);
 
     const task = new Task(
-        taskName,
-        taskDescription,
-        getDurationMinutes(taskLength, taskLengthUnit),
-        taskPriority);
-    allTaskJson.push(task);
+        taskName, taskDescription,
```
Comment (jamesnam-google) [inline portfolio/src/main/webapp/task.js]: Consider calling one param per line.

File: portfolio/src/main/webapp/task.js [hunk 6]
```diff
@@ -63,16 +63,23 @@ function createNewTask() {
...
-      priority);
-  
+      name, description, getDurationMinutes(length, lengthUnit), priority);
+
   updateTaskList(newTask, lengthUnit);
 }
 
+/** Shows the "Task Added" header. */
+function showTaskAddedHeader() {
+  const $taskListHeader = $('#task-added-header');
```
Comment (jamesnam-google) [inline portfolio/src/main/webapp/task.js]: We can merge these lines into

$('#task-added-header').removeClass('d-none');

File: portfolio/src/main/webapp/request.js [hunk 7]
```diff
@@ -83,19 +94,17 @@ function collectAllScheduledTasks() {
...
     const scheduledTaskPriority =
         parseInt(cardBody.childNodes[4].getAttribute('data-task-priority'));
 
     var scheduledTask = {};
     const task = new Task(
-        scheduledTaskName,
-        scheduledTaskDescription,
-        scheduledTaskDurationMinutes,
-        scheduledTaskPriority);
+        scheduledTaskName, scheduledTaskDescription,
```
Comment (jamesnam-google) [inline portfolio/src/main/webapp/request.js]: Consider having one param per line.

**Review by jamesnam-google — APPROVED** [review]
Approving the changes with comments

File: portfolio/src/main/webapp/calendar.js (hunk 2 above)
Comment (raulcruise) [reply portfolio/src/main/webapp/calendar.js]: Done.

File: portfolio/src/main/webapp/task.js (hunk 6 above)
Comment (raulcruise) [reply portfolio/src/main/webapp/task.js]: Done.

File: portfolio/src/main/webapp/calendar.js (hunk 4 above)
Comment (raulcruise) [reply portfolio/src/main/webapp/calendar.js]: Done.

File: portfolio/src/main/webapp/task.js (hunk 5 above)
Comment (raulcruise) [reply portfolio/src/main/webapp/task.js]: Done.

File: portfolio/src/main/webapp/calendar.js (hunk 3 above)
Comment (raulcruise) [reply portfolio/src/main/webapp/calendar.js]: Done.

File: portfolio/src/main/webapp/request.js (hunk 7 above)
Comment (raulcruise) [reply portfolio/src/main/webapp/request.js]: Done.

File: portfolio/src/main/webapp/calendar.js (hunk 1 above)
Comment (raulcruise) [reply portfolio/src/main/webapp/calendar.js]: Done.

**Review by rgcase — APPROVED** [review]

**Pull request merged**